### DISCUSSION TITLE
NextJS redirect function is resulting in 404 in production

### DISCUSSION THREAD

**Discussion by zeptosky** [vercel/community#6762] [Help] [answered]
### Summary

The redirect function results in 404 no matter what page. NextJS 14.2.3

### Steps to Reproduce

I have a NextJS project deployed on vercel and using `redirect()` function in server action results in 404 in production but locally works fine. Refreshing after redirect renders the page with no issues. It only started to happen today (8 May 2024). I have not made any significant changes, just some zod validation adjustments, but even If I call redirect with nothing else like this: 
```
async function TestRedirect() {
  'use server';
  redirect('/dashboard');
}
...
<form action={TestRedirect}>
  <button type="submit">Redirect</button>
</form>
```
 it still gives 404. Doesn't even matter which page, they all seem to 404 initially.

The logs don't show any errors or warnings.

Please, where can I even begin with figuring out what the issue is?

**Comment by aleksa-codes** [comment, marked as answer]
Also happening on my side with server actions and a redirect.

@zeptosky Seems like it is related to a new Vercel CLI build again:
https://github.com/orgs/vercel/discussions/6651#discussioncomment-9236702

I tried rolling back to `VERCEL_CLI_VERSION=vercel@33.7.1` using the env variable and now the redirect works as expected. I hope we get a response and solution so we don't have to roll back to an old version.

**Reply by zeptosky** [reply]
Oh, thanks for the info, I thought I was going crazy.
